backend/experiment/rules/toontjehoger_4_absolute.py

```diff
@@ -184,3 +185,20 @@ def get_final_round(self, session):
         )
 
         return [*score, final, info]
+
+    def validate_playlist(self, playlist: Playlist):
+        errors = super().validate_playlist(playlist)
+
+        # Get group values from sections, ordered by group
+        groups = list(playlist.section_set.values_list('group', flat=True).order_by('group'))
+
+        # Check if the groups are sequential and unique
+        if groups != list(map(str, range(1, len(groups) + 1))):
+            errors.append(f"Groups in playlist sections should be sequential and unique from 1 to {len(groups)}. E.g. {list(range(1, len(groups) + 1))}")
+
+        # Check if the tags are 'a', 'b' or 'c'
+        tags = list(playlist.section_set.values_list('tag', flat=True).distinct())
```

Comment:
I'm surprised the test doesn't fail, because, testing with the actual playlist, this code gives me the following:
```python
QuerySet ['a', 'b', 'c', 'a', 'b', 'c', 'a', ...]>
```
You need to linclude a `order_by` clause (previous to `distinct` in the chain) in order to truly deduplicate values.

Reply:
Ah, I see, you didn't add a test with valid tags.

Reply:
The test with valid groups also has valid tags

Reply:
Also, it seems that distinct() does work without a group by. If I do this:

```py
# Check if the tags are 'a', 'b' or 'c'
tags = list(playlist.section_set.values_list('tag', flat=True).distinct())
print(tags, len(tags))
```

I get:

```sh
['a', 'b', 'c'] 3
['a', 'b', 'c', 'd'] 4
['a', 'b', 'c'] 3
```

Reply:
But did you try adding the actual playlist? If that works, I concede - even though I locally ran this code with the actual playlist and got a different result.

Reply:
Apparently `distinct` works differently in unit tests than in real life 🤯 I've added the `order_by` now